src/browser-locale/index.js

```diff
@@ -0,0 +1,64 @@
+/* global TERRA_AGGREGATED_LOCALES */
+
+// The fallback locale that will be used if a supported locale is unable to be retrieved from the browser.
+const DEFAULT_LOCALE = 'en';
+
+/**
+ * Returns a boolean indicating if the locale is supported.
+ * @param {string} locale - The locale.
+ * @returns {bool} - A boolean indicating the locale is supported.
+ */
+const isSupported = (locale) => {
+  if (typeof TERRA_AGGREGATED_LOCALES === 'object' && Array.isArray(TERRA_AGGREGATED_LOCALES) && locale) {
+    return TERRA_AGGREGATED_LOCALES.indexOf(locale) > -1;
+  }
+
+  return false;
+};
+
+/**
+ * Filters the user's preferred languages from the browser and returns the first language found that is included in the supported locale list.
+ * @returns {string|null} - A supported locale. Null if not found.
+ */
+const filterLanguages = () => {
+  if (navigator.languages && navigator.languages.length > 0) {
```

Comment:
not a big deal, but i believe we can simplify the second conditional to a length check
```suggestion
  if (navigator.languages && navigator.languages.length) {
```